libcalico-go/lib/backend/syncersv1/updateprocessors/workloadendpointprocessor.go

```diff
@@ -163,6 +163,20 @@ func convertWorkloadEndpointV2ToV1Value(val interface{}) (interface{}, error) {
 		}
 	}
 
+	var qosControls *model.QoSControls
+	if v3res.Spec.QoSControls != nil {
+		qosControls = &model.QoSControls{
```

Comment:
Is you make these the same type you can just copy the whole struct instead of going field-by-field.